builtin/providers/github/resource_github_membership.go

```diff
@@ -19,26 +22,94 @@ func resourceGithubMembership() *schema.Resource {
 				ForceNew: true,
 			},
 			"role": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				//TODO validate function (member, admin)
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateRoleValue,
 			},
 		},
 	}
 }
 
 func resourceGithubMembershipCreate(d *schema.ResourceData, meta interface{}) error {
-	return nil
+	client := meta.(*GithubClient).client
+	n := d.Get("username").(string)
+	r := d.Get("role").(string)
+
+	membership, _, err := client.Organizations.EditOrgMembership(n, meta.(*GithubClient).organization,
+		&github.Membership{Role: &r})
+	if err != nil {
+		return err
+	}
+
+	orgName := membership.Organization.Login
+	userName := membership.User.Login
+
+	d.SetId(buildMembershipId(*orgName, *userName))
+
+	return resourceGithubMembershipRead(d, meta)
 }
 
 func resourceGithubMembershipRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*GithubClient).client
+
+	membership, _, err := client.Organizations.GetOrgMembership(d.Get("username").(string), meta.(*GithubClient).organization)
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+	username := membership.User.Login
+	roleName := membership.Role
+
+	d.Set("username", *username)
+	d.Set("role", *roleName)
 	return nil
 }
 
 func resourceGithubMembershipUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*GithubClient).client
+	n := d.Get("username").(string)
+	r := d.Get("role").(string)
+
+	if d.HasChange("role") {
```

Comment:
I don't think you need to explicitly check for a change here. Since `username` is marked as `ForceNew` you wont' every enter this update function unless there is a change to the value of `role`.